[5.1] docs(charts): update trendline articles to include the new types (#1887)
telerik/blazor-docs: components/chart/trendlines.md

## Trendline Types

The Chart supports the following fundamental types of trendlines:

> 

Reviewer: Drop the monospace formatting in the list.

* [`Linear Trendline`](#linear-trendline)
* [`Moving Average Trendline`](#moving-average-trendline)
* [`Exponential Trendline`](#exponential-trendline)
* [`Logarithmic Trendline`](#logarithmic-trendline)
* [`Power Trendline`](#power-trendline)
* [`Polynomial Trendline`](#polynomial-trendline)

### Linear Trendline

> The `Linear Trendline` serves the purpose of indicating whether a specific quantity is on the rise or decline over time.

Reviewer: ```suggestion
Use the Linear Trendline to visualize the rise or decline of a specific quantity over time.
```

### Moving Average Trendline

> The `Moving Average Trendline` smoothens out data fluctuations by computing an average of all data points over a specified period. By default, this period is set to `2` chart intervals.

Reviewer: ```suggestion
Use the Moving Average Trendline to smoothen out data fluctuations. This trendline computes and visualizes an average of all data points over a specified period. By default, this period is set to two chart intervals.
```

}
`````

### Exponential Trendline

> The `Exponential Trendline` shows data with rapidly accelerating growth or decay over time. It's used to emphasize trends that follow an exponential pattern, and it requires positive `Y` values.

Reviewer: ```suggestion
Use the Exponential Trendline to visualize data with rapidly accelerating growth or decay over time. This trendline helps you to emphasize trends that follow an exponential pattern, and it requires positive `Y` values.
```

                Price = 84.25m
            },
            new StockPriceTrend()
            {
                Period = "Q2 2022",
                NumericPeriod = 140,
                Date = new DateTime(2022, 4, 1),
                Price = 85.4m
            },
            new StockPriceTrend()
            {
                Period = "Q3 2022",
                NumericPeriod = 150,
                Date = new DateTime(2022, 7, 1),
                Price = 79.93m
            },
            new StockPriceTrend()
            {
                Period = "Q4 2022",
                NumericPeriod = 160,
                Date = new DateTime(2022, 10, 1),
                Price = 80.76m
            },
            new StockPriceTrend()
            {
                Period = "Q1 2023",
                NumericPeriod = 170,
                Date = new DateTime(2023, 1, 1),
                Price = 128.36m
            },
        };
    }

    private class StockPriceTrend
    {
        public string Period { get; set; }
        public int NumericPeriod { get; set; }
        public DateTime Date { get; set; }
        public decimal Price { get; set; }
    }
}
`````

### Logarithmic Trendline

> The `Logarithmic Trendline` is applied to data that displays rapid initial growth that slows over time or vice versa. It emphasizes trends with a logarithmic pattern, and it requires positive `X` values.

Reviewer: ```suggestion
Use the Logarithmic Trendline to visualize data with rapid initial growth that slows over time or vice versa. This trendline emphasizes trends with a logarithmic pattern, and it requires positive `X` values.
```

                Price = 84.25m
            },
            new StockPriceTrend()
            {
                Period = "Q2 2022",
                NumericPeriod = 140,
                Date = new DateTime(2022, 4, 1),
                Price = 85.4m
            },
            new StockPriceTrend()
            {
                Period = "Q3 2022",
                NumericPeriod = 150,
                Date = new DateTime(2022, 7, 1),
                Price = 79.93m
            },
            new StockPriceTrend()
            {
                Period = "Q4 2022",
                NumericPeriod = 160,
                Date = new DateTime(2022, 10, 1),
                Price = 80.76m
            },
            new StockPriceTrend()
            {
                Period = "Q1 2023",
                NumericPeriod = 170,
                Date = new DateTime(2023, 1, 1),
                Price = 128.36m
            },
        };
    }

    private class StockPriceTrend
    {
        public string Period { get; set; }
        public int NumericPeriod { get; set; }
        public DateTime Date { get; set; }
        public decimal Price { get; set; }
    }
}
`````

### Power Trendline

> The `Power Trendline` is used for data that follows a power-law relationship, indicating that one variable's change is proportional to a power of another variable. It helps highlight trends where the rate of change isn't constant. The `Y` and `X` require positive values.

Reviewer: ```suggestion
Use the Power Trendline to visualize data that follows a power-law relationship, indicating that one variable's change is proportional to a power of another variable. This trendline helps you to highlight trends where the rate of change isn't constant. It requires positive `Y` and `X` values.
```

                Price = 84.25m
            },
            new StockPriceTrend()
            {
                Period = "Q2 2022",
                NumericPeriod = 140,
                Date = new DateTime(2022, 4, 1),
                Price = 85.4m
            },
            new StockPriceTrend()
            {
                Period = "Q3 2022",
                NumericPeriod = 150,
                Date = new DateTime(2022, 7, 1),
                Price = 79.93m
            },
            new StockPriceTrend()
            {
                Period = "Q4 2022",
                NumericPeriod = 160,
                Date = new DateTime(2022, 10, 1),
                Price = 80.76m
            },
            new StockPriceTrend()
            {
                Period = "Q1 2023",
                NumericPeriod = 170,
                Date = new DateTime(2023, 1, 1),
                Price = 128.36m
            },
        };
    }

    private class StockPriceTrend
    {
        public string Period { get; set; }
        public int NumericPeriod { get; set; }
        public DateTime Date { get; set; }
        public decimal Price { get; set; }
    }
}
`````

### Polynomial Trendline

> The `Polynomial Trendline` is applied to data with complex patterns that cannot be captured by simpler trendlines. It fits a polynomial equation to the data points, allowing for a more flexible representation of trends with multiple turning points.

Reviewer: ```suggestion
Use the Polynomial Trendline to visualize complex data patterns not fitting the other trendlines. This trendline fits a polynomial equation to the data points, allowing for a more flexible representation of trends with multiple turning points.
```

                Price = 84.25m
            },
            new StockPriceTrend()
            {
                Period = "Q2 2022",
                NumericPeriod = 140,
                Date = new DateTime(2022, 4, 1),
                Price = 85.4m
            },
            new StockPriceTrend()
            {
                Period = "Q3 2022",
                NumericPeriod = 150,
                Date = new DateTime(2022, 7, 1),
                Price = 79.93m
            },
            new StockPriceTrend()
            {
                Period = "Q4 2022",
                NumericPeriod = 160,
                Date = new DateTime(2022, 10, 1),
                Price = 80.76m
            },
            new StockPriceTrend()
            {
                Period = "Q1 2023",
                NumericPeriod = 170,
                Date = new DateTime(2023, 1, 1),
                Price = 128.36m
            },
        };
    }

    private class StockPriceTrend
    {
        public string Period { get; set; }
        public int NumericPeriod { get; set; }
        public DateTime Date { get; set; }
        public decimal Price { get; set; }
    }
}
`````

> ## Chart Trendlines Parameters

Reviewer 2: I believe it makes sense to move the theory above the examples.
